pyiron_atomistics/vasp/base.py

```diff
@@ -1842,6 +1860,20 @@ def __init__(self):
         self.potcar = Potcar(table_name="potcar")
 
         self._eddrmm = "warn"
+        self._allow_structure_reordering = True
+
+    @property
+    def allow_structure_reordering(self) -> bool:
+        """
+
+        Returns:
+            bool
+        """
+        return self.input.allow_structure_reordering
+
+    @allow_structure_reordering.setter
+    def allow_structure_reordering(self, val: bool):
+        self.input.allow_structure_reordering = val
```

Comment:
I think that creating another flag level makes sense for expert users. Although I don't have the authority or the final say on these things. Let's just implement this for now because I am very reluctant to start large-scale segregation calculations without this flag. You could say I'm in a little bit a hurry x)

I'm fond of a name like input.options, it's an informative name and people will know where to look if they're looking for more granular control of the jobs.

Reply:
We're on the same page for the ideal situation then, and I appreciate your need for haste, but haste is dangerous in two ways here:

1) Once we include `vasp.input.allow_structure_reordering` in a release, we have an obligation to provide backwards compatibility to this interface for a while going forward. This is true even if we quickly switch to something like `vasp.input.options.allow_structure_reordering`. This clutters the code and adds maintenance work, so if we can avoid it by choosing a good name and design to start with that is much better.

2) On closer reading I realized your current PR doesn't do what you want. You would have found that you shipped all your jobs off to the queue and they came back reordered anyhow. The reason for this is that `Input.allow_structure_reordering` gets declared `True` in `__init__`, but never gets serialized to HDF. Our execution pattern is to serialize the local object, then re-initialize and deserialize it on the remote python instance. So over on the compute node the job would only ever see the initial value! The correct solution to this is adding tests that ensure that vasp job (de)serialization is working correctly with your new changes.

I made a new PR based off this one, #799, which introduces the input sub-domain discussed above and (I believe) resolves the serialization issues. It still needs tests. My proposal is that you track down someone at MPIE to sit with you and write the tests ASAP, and that we have a quick discussion of how to name `job.input.NAMETHISTHING.allow_structure_reordering` on Monday -- is this fast enough for your timeline?

Reply:
I think it's fine - I can hack something on a modified module that I'll use for my personal purposes - Putting the pyiron_atomistics folder in the folder where I'm working and modifying that is fine. 

For something that is more sustainable it is probably better that we write a solution that mostly everyone is happy with maintaining going forward

Reply:
I'll have a look at the tests on the weekend. But honestly the talk of serialisation and all that flies over my head - explain like I'm 5? I will probably need it explained before I know what I'm testing for.

I can probably grab Sam tomorrow to explain, since Marvin is off at a conference.

Reply:
> I'll have a look at the tests on the weekend. But honestly the talk of serialisation and all that flies over my head - explain like I'm 5? I will probably need it explained before I know what I'm testing for.
> 
> I can probably grab Sam tomorrow to explain, since Marvin is off at a conference.

Super.

Yeah, of course! "Serialization" here just means writing to file. Basically when you're working with the job in a notebook, it exists in RAM using your current python instance. To run it on the remote node, we start a new python instance utilizing the node's RAM and re-instantiate the job. That means that the re-instantiated job needs some way to access all the modifications you made from the clean, initialized state (settings changes, etc). These are what's read from file, i.e. "deserialization". So in this case the meat of your test might look something like this:

```python
original_job = pr.atomistics.job.Vasp('testing_serialization')
original_job.input.code_specific_options.allow_structure_reordering = False
# Maybe also play with kpoints, etc. if we don't have any tests for these either
original_job.to_hdf()

loaded_job = pr.atomistics.job.Vasp(original_job.name)
# loaded_job and original_job should now be separate python objects with different hashes
# At this point, if you print out allow_structure_reordering, it will still have the initial value of True
loaded_job.from_hdf()
self.assertEquals(
    original_job.input.code_specific_options.allow_structure_reordering,
    loaded_job.input.code_specific_options.allow_structure_reordering
)
```

I might have made some syntax errors there, and I'm assuming you're inside a `TestCase.TEST_SOMETHING` and have access to a `pr` of type `Project`, etc, but I hope the logic/intent is sufficiently clear.